docs/tutorials/writing_tutorials.md

```diff
@@ -0,0 +1,307 @@
+error at the first failing commands.
+
+##### `"HereTag":"EOF"`
+
+In conjunction with a bash runtime, you can specify that the content contains a
+Here Tag. All the lines until a line starting by EOF will be taken into the
+script.
+
+This chunk must only have a single command in it.
+
+##### `"label":"some label"`
+
+You can give a pretty name to a chunk. It's good for bash runtimes, as the
+command to execute is always `./script.sh` so if you want to have a
+differentiable name in the logs, use this field.
+
+##### `"variables":["NAME"]`
+
+Extracts the given variables from the environment of the chunk. This works only
+in conjunction with a bash environment. User can use this to either extract
+content they are creating or system variables, such as EDITOR for instance.
+
+##### `"env":["NAME=value"]`
+
+Gives environment variables to all the commands of the chunk.
+
+##### `"rootdir":"$operator"`
+
+The commands in the chunk will the executed from the operator source code
+directory.
+
+
+##### `"rootdir":"$tmpdir.x"`
+
+Create a new temporary directory to execute the chunk. If another chunk reuses
+the same suffix (`.x` in the example) it'll share the same directory. Useful
+when you need to chain chunks or to reuse some cached files between chunks
+
+##### `"parallel":"true"`
+
+Makes the chunk executed in parallel of the others in its stage. This works best
+if there's only one command in the chunk (or if the chunk is a bash chunk). And
+if all the commands in the stage are made to run in parallel.
+
+For now the parallel behavior has a simplistic implementation. Only the last
+command of the chunk is really started asynchronously. The other ones are
+sequentially executed before that.
+
+##### `"breakpoint":"true"`
+
+Enters interactive mode when the chunk is started. Useful for debugging
+purposes. Better to use alongside `--verbose`
+
+##### `"requires":"stageName/id"`
+
+Makes the execution of the given stage dependant of the correct execution of the
+pointed stage. To be used in teardown chunks.
+
+##### `"id":"someID"`
+
+Give an ID to a chunk so that it can get referenced later on
+
+#### Commands
+
+Commands start with a `$` in front of them
+
+### Examples
+
+````
+```{"stage":"test1"}
+$ echo a line starting with dollar gets executed, and its output is shown under
+
+a line starting with dollar gets executed, and its output is shown under
+```
+````
+
+Running the `go run test/utils/tutorials/tester.go` command will execute
+the above chunk and produce:
+
+````
+```
+╼> go run test/utils/tutorials/tester.go --run-only writing_tutorials.md
+
+# Testing writing_tutorials.md
+
+ SUCCESS  echo a line starting with dollar gets executed, and its output is shown under
+```
+````
+
+(Note that this chunk isn't executed as there's no JSON metadata associated)
+
+Let's have a verbose view:
+
+```
+╼> go run test/utils/tutorials/tester.go --verbose --run-only writing_tutorials.md
```

Comment:
The option `--run-only` seems more a way to test tutorials than a way to select a tutorial, the test goal of the maven surefire plugin has a parameter with the name `test` to define the test to run, see https://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html

Reply:
Yes I need a way to execute a single test at a time. I couldn't find a way to have positional arguments with the golang flag package easily, that's why I ended up using this option. You recommend to rename the `run-only` to `test`?

Reply:
Also `tutorial` could be a good candidate, WDYT?

Reply:
I think that's way better than `run-only` thanks!